### DISCUSSION TITLE
Is there any documentation of the required permissions for each API endpoint?

### DISCUSSION THREAD

**Discussion by drmason13** [CrowdStrike/falconpy#1117] [Q&A] [answered]
Hello Falconpy community!

I'm sure loads of you have managed to setup permissions just fine, possibly with a bit of trial and error.

But is there anywhere that records the permissions required to perform each API action?

Specifically I think I'm looking for permissions to get _detections_.

If there was also a chance to get pointed to how to configure those permissions in Crowdstrike I'd appreciate it, thank you.

**Comment by jshcodes** [comment, marked as answer]
Hi @drmason13 -

For the [`QueryDetects`](https://falconpy.io/Service-Collections/Detects.html#querydetects) or [`GetDetectSummaries`](https://falconpy.io/Service-Collections/Detects.html#getdetectsummaries) operations, you will need the Detects __READ__ scope.  This can be configured for your API client within the Falcon Console. (Menu (upper left) -> Support & Resources -> API clients and keys).

Regarding providing this detail for all operations, this [request is in flight](https://github.com/CrowdStrike/falconpy/issues/1087). This detail should be showing up on [falconpy.io](https://falconpy.io/Service-Collections/Detects.html) soon.  😄
